From: Tomasz Rybak <serpent@debian.org>
Date: Tue, 24 Sep 2024 21:12:38 +0200
Subject: Ensure reproducible documentation

Forwarded: not-needed
Last-Update: 2024-12-20

Do not fetch external Sphinx inventory but use one from Debian packages.
===================================================================
---
 docs/conf.py           | 9 ++++++---
 docs/lib/libpq_docs.py | 4 +++-
 2 files changed, 9 insertions(+), 4 deletions(-)

diff --git a/docs/conf.py b/docs/conf.py
index cac9736..af8ab0a 100644
--- a/docs/conf.py
+++ b/docs/conf.py
@@ -98,9 +98,12 @@ html_static_path = ["_static"]
 default_role = "obj"
 
 intersphinx_mapping = {
-    "py": ("https://docs.python.org/3", None),
-    "pg2": ("https://www.psycopg.org/docs/", None),
-    "numpy": ("https://numpy.org/doc/stable/", None),
+    "py": ("file:///usr/share/doc/python3-doc/html/",
+        "/usr/share/doc/python3-doc/html/objects.inv"),
+    "pg2": ("file:///usr/share/doc/python-psycopg2-doc/html/",
+        "/usr/share/doc/python-psycopg2-doc/html/objects.inv"),
+    "numpy": ("file:///usr/share/doc/python-numpy/html/",
+        "/usr/share/doc/python-numpy/html/objects.inv"),
 }
 
 autodoc_member_order = "bysource"
diff --git a/docs/lib/libpq_docs.py b/docs/lib/libpq_docs.py
index a5488d1..f105089 100644
--- a/docs/lib/libpq_docs.py
+++ b/docs/lib/libpq_docs.py
@@ -8,6 +8,8 @@ Add the ``:pq:`` role, to create a link to a libpq function, e.g. ::
 
 will link to::
 
+    file:///usr/share/doc/postgresql-doc-16/html/libpq-misc.html #LIBPQ-PQLIBVERSION
+previously to:
     https://www.postgresql.org/docs/current/libpq-misc.html #LIBPQ-PQLIBVERSION
 
 """
@@ -82,7 +84,7 @@ class LibpqParser(HTMLParser):
             func_id=self.varlist_id.upper(),
         )
 
-    _url_pattern = "https://www.postgresql.org/docs/{version}/{section}.html#{func_id}"
+    _url_pattern = "file:///usr/share/doc/postgresql-doc-{version}/html/{section}.html#{func_id}"
 
 
 class LibpqReader:
